internal/pkg/describe/env.go

```diff
@@ -62,41 +63,63 @@ func NewEnvDescriber(opt NewEnvDescriberConfig) (*EnvDescriber, error) {
 
 		configStore:    opt.ConfigStore,
 		deployStore:    opt.DeployStore,
-		stackDescriber: stackAndResourcesDescriber(d),
+		stackDescriber: d,
 	}, nil
 }
 
 // Describe returns info about an application's environment.
-func (e *EnvDescriber) Describe() (*EnvDescription, error) {
-	svcs, err := e.filterDeployedSvcs()
+func (d *EnvDescriber) Describe() (*EnvDescription, error) {
+	svcs, err := d.filterDeployedSvcs()
 	if err != nil {
 		return nil, err
 	}
 
-	tags, err := e.stackTags()
+	tags, err := d.stackTags()
 	if err != nil {
 		return nil, fmt.Errorf("retrieve environment tags: %w", err)
 	}
 
 	var stackResources []*CfnResource
-	if e.enableResources {
-		stackResources, err = e.envOutputs()
+	if d.enableResources {
+		stackResources, err = d.envOutputs()
 		if err != nil {
 			return nil, fmt.Errorf("retrieve environment resources: %w", err)
 		}
 	}
 
 	return &EnvDescription{
-		Environment: e.env,
+		Environment: d.env,
 		Services:    svcs,
 		Tags:        tags,
 		Resources:   stackResources,
 	}, nil
 }
 
-func (e *EnvDescriber) stackTags() (map[string]string, error) {
+// Version returns the CloudFormation template version associated with
+// the environment by reading the Metadata.Version field from the template.
+//
+// If the Version field does not exist, then it's a legacy template and it returns an empty string and nil error.
+func (d *EnvDescriber) Version() (string, error) {
+	body, err := d.stackDescriber.Template(stack.NameForEnv(d.app, d.env.Name))
+	if err != nil {
+		return "", err
+	}
+
+	type metadata struct {
+		Version string `yaml:"Version"`
+	}
+	tpl := struct {
+		Metadata metadata `yaml:"Metadata"`
+	}{}
+	if err := yaml.Unmarshal([]byte(body), &tpl); err != nil {
+		return "", fmt.Errorf("unmarshal CloudFormation template to read Version: %w", err)
```

Comment:
Do we want to return an empty string or return a special version number like "v0.0.0" if they are using the legacy version? It might not be explicit that legacy version will return "" as a version number...

Reply:
Done! good idea